clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6943,15 +6943,70 @@ void OffloadWrapper::ConstructJob(Compilation &C, const JobAction &JA,
     WrapperArgs.push_back(
         C.getArgs().MakeArgString(Twine("-kind=") + Twine(Kind)));
 
+    ArgStringList ForeachArgs;
+
     for (const InputInfo &I : Inputs) {
       assert(I.isFilename() && "Invalid input.");
-      WrapperArgs.push_back(I.getFilename());
+      std::string FileName(I.getFilename());
+      if (I.getType() == types::TY_Tempfilelist ||
+          I.getType() == types::TY_TempEntriesfilelist) {
+        ForeachArgs.push_back(
+            C.getArgs().MakeArgString("--in-file-list=" + FileName));
+        ForeachArgs.push_back(
+            C.getArgs().MakeArgString("--in-replace=" + FileName));
```

Comment:
If it's so often the case that `--*-file-list=` and `--*-replace=` values match, why not just provide a single option?

I guess this question is directed more so at `llvm-foreach`, which is not the focus of this review, but still.

Reply:
It's often the case only here, in the Driver, and It's some kind of lazy solution. I think normally people would give different values for these options.
I think we also can give different values for these options here if we need it, It can simplify the picture when you look at `-###` output, for example.

Reply:
Moving the related discussion to the `constructLLVMForeachCommand()` area